core/mainstage.go

```diff
@@ -165,17 +165,23 @@ func (st *Stage) firstWindowStages() *stages {
 
 // configMainStage does main-stage configuration steps
 func (st *Stage) configMainStage() {
+	sc := st.Scene
 	if st.NewWindow {
 		st.FullWindow = true
 	}
+	if st.NewWindow && st.Type == DialogStage && TheApp.Platform().IsMobile() && st.Context != nil && st.Context.AsWidget().SizeClass() != SizeCompact {
+		st.NewWindow = false
+		st.FullWindow = false
+		st.Modal = false
+		st.Scrim = false
+	}
```

Comment:
You can fold this if statement into the end of the if statement below to get rid of the duplicate mobile condition and `NewWindow` setting.

Reply:
easier if you just do the logic the way you want it.